src/player.cpp

```diff
@@ -10896,15 +10925,17 @@ bool player::wield_contents( item &container, int pos, bool penalties, int base_
         return false;
     }
 
+    std::string err;
     auto target = std::next( container.contents.begin(), pos );
-    if( !can_wield( *target ) ) {
+    if( !can_wield( *target, &err ) ) {
+        add_msg_if_player( m_info, "%s", err.c_str() );
         return false;
     }
 
     int mv = 0;
 
     if( is_armed() ) {
-        if( !wield( ret_null ) ) {
+        if( !wield( weapon ) ) {
```

Comment:
Wielding currently wielded item to unwield it is certainly more hacky than wielding a lack of item to unwield current item.

Reply:
Agreed. Will change that.

Reply:
Done.